lib/pat.c

```diff
@@ -5876,26 +5887,42 @@ search_push(grn_ctx *ctx,
       }
     }
   } else {
-    pat_node *pn;
-    PAT_AT(pat, id, pn);
-    if (pn) {
-      int32_t ch = PAT_CHK(pn);
+    pat_node_common *node;
+    PAT_AT(pat, id, node);
+    if (node) {
+      int32_t check = pat_node_get_check(pat, node);
+      // operation of "key_len * 16" is for comparing check and key_len.
+      //
+      // check is pat_node.check or pat_node_large.check.
+      // check include "nth byte", "nth bit", "terminated".
+      // eg) check = 190
+      // 0b011111010000 000      0
+      //   ^nth byte    ^nth bit ^terminated
+      // If we want to extract nth byte from check, execute (check >> 4).
+      // Therefore, to compare against key_len you can either:
+      //
+      // 1. Compare check and key_len:
+      // check == key_len * 16
+      //
+      // 2. Compare nth byte and key_len:
+      // (check >> 4) == key_len *16
       int32_t len = key_len * 16;
```

Comment:
Should we use `PAT_CHECK_PACK(ken_len, 0, 0)` instead of `key_len * 16`, and more meaningful variable name than `len`?

Reply:
Thank you very much.
I used `PAT_CHECK_PACK(ken_len, 0, flase)`.